### PR TITLE
W-10915280:Imports failing in AST validation

### PR DESCRIPTION
**Pull request by egrave**

### CODE REVIEW

**Review by mpereyra-salesforce — COMMENTED** [review]
It looks right and it compiles.
I only found an only possible enhancement. Let me know if you think could be achieved and I will approve this.

File: mule-extension-model-loader/src/main/java/org/mule/tooling/api/AstGenerator.java [hunk 1]
```diff
@@ -84,7 +98,7 @@ public ArtifactAst generateAST(List<String> configs, Path configsPath) throws Fi
   }
 
   public ArrayList<ValidationResultItem> validateAST(ArtifactAst artifactAst) throws ConfigurationException {
-    ValidationResult result = MuleAstUtils.validate(artifactAst);
+    ValidationResult result = MuleAstUtils.validatorBuilder().build().validate(artifactAst);
     ArrayList<ValidationResultItem> errors = new ArrayList<ValidationResultItem>();
     ArrayList<ValidationResultItem> warnings = new ArrayList<ValidationResultItem>();
     result.getItems().forEach(v ->{if(v.getValidation().getLevel().equals(Level.ERROR)){errors.add(v);}else {warnings.add(v);}});
```
Comment (rodriguez-jorge) [inline mule-extension-model-loader/src/main/java/org/mule/tooling/api/AstGenerator.java:104]: Maybe using  `result.getItems().stream()collect(groupingBy(v -> v.getValidation().getLevel()))` is more readable.

File: mule-extension-model-loader/src/main/java/org/mule/tooling/internal/DefaultExtensionModelService.java [hunk 2]
```diff
@@ -335,16 +336,19 @@ private void createPomFile(BundleDescriptor pluginDescriptor, String uuid, File
...
+      ArrayList<URL> resources = new ArrayList<URL>();
+      artifactPluginDescriptor.getClassLoaderModel().getExportedResources().forEach(resource->
+      resources.add(toolingArtifactClassLoader.getRegionClassLoader().getResource(resource)));
       MuleExtensionModelLoaderManager extensionModelLoaderRepository =
           new MuleExtensionModelLoaderManager(muleArtifactResourcesRegistry.getContainerArtifactClassLoader());
       extensionModelLoaderRepository.start();
       final Set<Pair<ArtifactPluginDescriptor, ExtensionModel>> loadedExtensionInformation =
           discoverPluginsExtensionModel(toolingArtifactClassLoader, extensionModelLoaderRepository, properties);
-      return loadedExtensionInformation;
+      return new PluginResources(loadedExtensionInformation,resources);
```
Comment (rodriguez-jorge) [inline mule-extension-model-loader/src/main/java/org/mule/tooling/internal/DefaultExtensionModelService.java:351]: Please add an space `, resources`

File: mule-extension-model-loader/src/main/java/org/mule/tooling/internal/PluginResources.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import java.util.Set;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.util.Pair;
+import org.mule.runtime.deployment.model.api.plugin.ArtifactPluginDescriptor;
+
+public class PluginResources {
+
+  Set<Pair<ArtifactPluginDescriptor, ExtensionModel>> extensionModels;
+  List<URL> exportedResources;
```
Comment (rodriguez-jorge) [inline mule-extension-model-loader/src/main/java/org/mule/tooling/internal/PluginResources.java:14]: Is the visibility of the properties correct? can the properties be final?

**Review by rodriguez-jorge — COMMENTED** [review]
Only minor comments. The only doubt is the properties visibility.

**Review by rodriguez-jorge — APPROVED** [review]
LGTM

File: mule-extension-model-loader/src/main/java/org/mule/tooling/api/AstGenerator.java (hunk 1 above)
Comment (egrave) [reply mule-extension-model-loader/src/main/java/org/mule/tooling/api/AstGenerator.java:104]: This is not part of this PR. Nevertheless I tried that line and it didn't work for me ( but I have no experience with the groupingBy). Maybe we can review this in the future

**Pull request merged**